src/classes/OPP_OpportunityContactRoles_TEST.cls

```diff
@@ -879,5 +879,67 @@ private class OPP_OpportunityContactRoles_TEST {
         Test.stopTest();
     }
 
+    /*******************************************************************************************************
+    * @description Validates that changing the Opp.AccountId to a null value or to a different Account does not
+    * cause an error and that the related OCR's are valid
+    */
+    static testMethod void testOppAccountIdChanges() {
+        //skip the test if Advancement is installed
+        if(ADV_PackageInfo_SVC.useAdv()) return;
+
+        Account orgAcct = new Account(Name = 'OrgTest567890', npe01__SYSTEMIsIndividual__c = false);
+        insert orgAcct;
+
+        npe01__Contacts_and_Orgs_Settings__c testSettings = UTIL_CustomSettingsFacade.getContactsSettingsForTests(
+                new npe01__Contacts_and_Orgs_Settings__c(
+                        npe01__Account_Processor__c = CAO_Constants.HH_ACCOUNT_PROCESSOR,
+                        npe01__Opportunity_Contact_Role_Default_role__c = OPPORTUNITY_CONTACT_ROLE_DONOR,
+                        Honoree_Opportunity_Contact_Role__c = OPPORTUNITY_CONTACT_ROLE_HONOREE,
+                        Notification_Recipient_Opp_Contact_Role__c = OPPORTUNITY_CONTACT_ROLE_NOTIFICATION_RECIPIENT
+                )
+        );
+
+        npo02__Households_Settings__c householdSettingsForTests = UTIL_CustomSettingsFacade.getHouseholdsSettingsForTests(
+                new npo02__Households_Settings__c (
+                        npo02__Household_Rules__c = HH_Households.NO_HOUSEHOLDS_PROCESSOR,
+                        npo02__Household_Contact_Roles_On__c = true,
+                        npo02__Household_Member_Contact_Role__c='Household Member'
+                )
+        );
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        Contact c = [SELECT AccountId FROM Contact WHERE Id = :con.id LIMIT 1];
+
+        Opportunity opp1 = new Opportunity(
+                Name = 'Apex Test Opp 3',
+                Primary_Contact__c = c.Id,
+                CloseDate = date.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage()
+        );
+        insert opp1;
+
+        opp1 = [SELECT Id, AccountId FROM Opportunity WHERE Id = :opp1.Id LIMIT 1];
+        System.assertNotEquals(null, opp1.AccountId);
+
+        Test.startTest();
+        opp1.AccountId = null;
+        update opp1;
+        OpportunityContactRole[] result = [SELECT OpportunityId, Opportunity.AccountId, ContactId, isPrimary, Role FROM OpportunityContactRole WHERE OpportunityId = :opp1.Id];
+        //should be a contact role
+        System.assertEquals(1, result.size());
+        System.assertEquals(OPPORTUNITY_CONTACT_ROLE_DONOR, result[0].Role);
+        System.assertEquals(true, result[0].isPrimary);
+
```

Comment:
@force2b -- A few items:
1. Add `System.assertEquals(null, result[0].Opportunity.AccountId);` after line 933
2. Formatting suggestion -- Change from `OpportunityContactRole[]` to `List<OpportunityContactRole>`
3. Change the variable name from `result` to `results` since the data type is a List/Array.

None of the above items are showstoppers.

Reply:
@joy-sfdo Very good points. I copied an existing test method and changed only a couple of items. I'm also not a fan of the pattern of using `[]` versus `List<>` or querying a single record into a list and then references as `[0]`.  As much as I really want to fix all those items, I think we're in a time crunch given how long builds take so if you don't mind I think we should merge as is.

Reply:
@force2b -- Sounds good, given the time crunch.